core/services/ocr2/plugins/ocr2keeper/evm21/feed_lookup.go

```diff
@@ -72,39 +73,38 @@ type AdminOffchainConfig struct {
 }
 
 // feedLookup looks through check upkeep results looking for any that need off chain lookup
-func (r *EvmRegistry) feedLookup(ctx context.Context, upkeepResults []ocr2keepers.CheckResult) ([]ocr2keepers.CheckResult, error) {
+func (r *EvmRegistry) feedLookup(ctx context.Context, checkResults []ocr2keepers.CheckResult) []ocr2keepers.CheckResult {
 	lookups := map[int]*FeedLookup{}
-	for i := range upkeepResults {
-		res := &upkeepResults[i]
+	for i, res := range checkResults {
 		if res.IneligibilityReason != uint8(UpkeepFailureReasonTargetCheckReverted) {
 			continue
 		}
 
 		block := big.NewInt(int64(res.Trigger.BlockNumber))
 		upkeepId := res.UpkeepID
 
-		opts, err := r.buildCallOpts(ctx, block)
-		if err != nil {
-			r.lggr.Errorf("[FeedLookup] upkeep %s block %d buildCallOpts: %v", upkeepId, block, err)
-			return nil, err
-		}
+		opts := r.buildCallOpts(ctx, block)
 
-		allowed, err := r.allowedToUseMercury(opts, upkeepId.BigInt())
+		state, retryable, allowed, err := r.allowedToUseMercury(opts, upkeepId.BigInt())
 		if err != nil {
-			r.lggr.Errorf("[FeedLookup] upkeep %s block %d failed to time mercury allow list: %v", upkeepId, block, err)
+			r.lggr.Warnf("[FeedLookup] upkeep %s block %d failed to query mercury allow list: %s", upkeepId, block, err)
```

Comment:
you can create a logger with some context to be used all over this function, instead of having a prefix in each log:
`lggr := r.lggr.With("where", "FeedLookup")`